tests/python/gpu/test_kvstore_gpu.py

```diff
@@ -59,13 +59,17 @@ def init_kv_with_str(stype='default', kv_type='local'):
 # Not reproducible, so this test is back on random seeds.
 @with_seed()
 def test_rsp_push_pull():
+    num_gpus = len(list_gpus())
+    if num_gpus < 2:
```

Comment:
I would do this in setUp, before the test is started and issue a warning

Reply:
Where is setUp? this is not the testing style used in the file, warning is fine with me.

Reply:
You would need to do something like:
```python
class TestNCCL(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        num_gpus = mx.context.num_gpus()
        if num_gpus < 2:
            raise unittest.SkipTest("It makes sense to test rsp_push_pull functionality on more than 1 GPU only")
```